addon-manager/api/src/main/java/org/jboss/forge/addon/manager/catalog/AbstractAddonDescriptorCatalog.java

```diff
@@ -0,0 +1,95 @@
+package org.jboss.forge.addon.manager.catalog;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+
+import org.jboss.forge.addon.manager.catalog.AddonDescriptor.AddonDescriptorCategory;
+
+/**
+ * AbstractAddonDescriptorCatalog
+ */
+public abstract class AbstractAddonDescriptorCatalog implements AddonDescriptorCatalog {
```

Comment:
Another round of design changes: How about making this class concrete and calling it `URLAddonDescriptorCatalog`?